src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1281,6 +1304,186 @@ internal BoundExpression BindCollectionBuilderCreate(
+                                    parameters: [("capacity", intType), ("comparer", comparerType)],
+                                    returnType,
+                                    diagnostics);
+                            }
+                        }
+                        break;
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(collectionTypeKind);
+                }
+            }
+
+            static void addSignature(
+                ArrayBuilder<MethodSymbol> builder,
+                CSharpCompilation compilation,
+                SyntaxNode syntax,
+                WellKnownMember wellKnownConstructor,
+                NamedTypeSymbol containingType,
+                string methodName,
+                ImmutableArray<(string Name, TypeWithAnnotations Type)> parameters,
+                TypeWithAnnotations returnType,
+                BindingDiagnosticBag diagnostics)
+            {
+                var constructor = (MethodSymbol?)GetWellKnownTypeMember(compilation, wellKnownConstructor, diagnostics, syntax: syntax);
+                if (constructor is { })
+                {
+                    var method = new CollectionArgumentsSignatureOnlyMethodSymbol(constructor, methodName, containingType, parameters, returnType);
+                    builder.Add(method);
+                }
+            }
+        }
+
+        private BoundExpression BindInterfaceTargetCollectionArguments(
+            SyntaxNode syntax,
+            BoundExpression methodGroup,
+            BoundCollectionExpressionWithElement? withElement,
+            TypeSymbol targetType,
+            BindingDiagnosticBag diagnostics)
+        {
+            Debug.Assert(methodGroup.Kind is BoundKind.MethodGroup or BoundKind.BadExpression);
+            if (methodGroup is BoundMethodGroup group)
+            {
+                var analyzedArguments = AnalyzedArguments.GetInstance();
+                withElement?.GetArguments(analyzedArguments);
+                var result = BindMethodGroupInvocation(
+                    syntax,
+                    expression: syntax,
+                    group.Name,
+                    group,
+                    analyzedArguments,
+                    diagnostics,
+                    queryClause: null,
+                    ignoreNormalFormIfHasValidParamsParameter: false,
+                    out _,
+                    disallowExpandedNonArrayParams: true,
+                    acceptOnlyMethods: true).MakeCompilerGenerated();
+                analyzedArguments.Free();
+                Debug.Assert(result.Kind != BoundKind.DynamicInvocation);
+                Debug.Assert(targetType.Equals(result.Type, TypeCompareKind.AllIgnoreOptions));
+                return result;
+            }
+            else
+            {
+                return new BoundBadExpression(syntax, LookupResultKind.Empty, symbols: [], childBoundNodes: [methodGroup], targetType);
```

Comment:
>[methodGroup] [](http://example.com/codeflow?start=108&length=13)

In error cases, the bad expression for method group doesn't capture any `childBoundNodes`. 
It feels like we should capture the with-arguments in the bad expression for the method group.
Then I"m not sure if we need this second bad expression on top of it.

Reply:
Good catch, thanks.

Reply:
nit: why do you prefer stacking two BoundBadExpression nodes, rather than returning one complete one from `BindInterfaceTargetCollectionMethodGroup`?

Reply:
Are we still stacking two `BoundBadExpression` nodes, after I changed `childBoundNodes` here? We could simple return one complete `BoundBadExpression` from `BindInterfaceTargetCollectionMethodGroup` but that would mean passing `withElement` in to that method, simply to determine the method group.